validator/db/kv/migration_optimal_attester_protection_test.go

```diff
@@ -0,0 +1,107 @@
+package kv
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/snappy"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/testutil/assert"
+	"github.com/prysmaticlabs/prysm/shared/testutil/require"
+	bolt "go.etcd.io/bbolt"
+)
+
+func Test_migrateOptimalAttesterProtection(t *testing.T) {
+	tests := []struct {
+		name  string
+		setup func(t *testing.T, db *bolt.DB)
+		eval  func(t *testing.T, db *bolt.DB)
+	}{
+		{
+			name: "only runs once",
+			setup: func(t *testing.T, db *bolt.DB) {
+				err := db.Update(func(tx *bolt.Tx) error {
+					return tx.Bucket(migrationsBucket).Put(migrationOptimalAttesterProtectionKey, migrationCompleted)
+				})
+				require.NoError(t, err)
+			},
+			eval: func(t *testing.T, db *bolt.DB) {
+				err := db.View(func(tx *bolt.Tx) error {
+					data := tx.Bucket(migrationsBucket).Get(migrationOptimalAttesterProtectionKey)
+					require.DeepEqual(t, data, migrationCompleted)
+					return nil
+				})
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "populates optimized schema buckets",
```

Comment:
maybe it can be good to add a partial data migration. were half of the data is already in the db but complete flag isn't set. check if migration completes